service/controller/internal/vmsku/vmsku.go

```diff
@@ -0,0 +1,107 @@
+package vmsku
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/azure-operator/v4/client"
+)
+
+const (
+	// CapabilitySupported is the value returned by this API from Azure when the capability is supported
+	CapabilitySupported = "True"
+
+	CapabilityAcceleratedNetworking = "AcceleratedNetworkingEnabled"
+)
+
+type Config struct {
+	ClientFactory *client.Factory
+	Location      string
+	Logger        micrologger.Logger
+}
+
+type Interface struct {
+	clientFactory *client.Factory
+	location      string
+	skus          map[string]*compute.ResourceSku
+	logger        micrologger.Logger
+}
+
+func New(config Config) (*Interface, error) {
+	if config.ClientFactory == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ClientFactory must not be empty", config)
+	}
+	if config.Location == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Location must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	return &Interface{
+		clientFactory: config.ClientFactory,
+		location:      config.Location,
+		logger:        config.Logger,
+	}, nil
+}
+
+func (v *Interface) HasCapability(ctx context.Context, vmType string, name string) (bool, error) {
+	if len(v.skus) == 0 {
+		err := v.initCache(ctx)
+		if err != nil {
+			return false, microerror.Mask(err)
+		}
+	}
+	vmsku, found := v.skus[vmType]
+	if !found {
+		return false, microerror.Maskf(skuNotFoundError, vmType)
+	}
+	if vmsku.Capabilities != nil {
+		for _, capability := range *vmsku.Capabilities {
+			if capability.Name != nil && *capability.Name == name {
+				if capability.Value != nil && strings.EqualFold(*capability.Value, string(CapabilitySupported)) {
+					return true, nil
+				}
+			}
+		}
+	}
+	return false, nil
+}
+
+func (v *Interface) initCache(ctx context.Context) error {
+	v.logger.LogCtx(ctx, "level", "debug", "message", "Initializing cache for VMSKU")
+	cl, err := v.clientFactory.GetResourceSkusClient("giantswarm", "credential-default")
```

Comment:
Is there a way to get the location of this secret in a better way?

Reply:
I believe we attempt to get in from the org namespace, and then fallback to the default one, e.g. https://github.com/giantswarm/azure-operator/blob/master/client/organization_factory.go#L178-L180

Reply:
here I don't really care about the organization. The outcome of this module is shared between all organizations (VM sku details do not change within the same region).
So I am really ok using the default credentials here.

Reply:
So there are two ways of using credentials in the operator. You either use the control plane credentials, or the tenant cluster credentials.

## Control plane creds

These are easy. They are built at the very beginning and just passed around. If you want to use these, just wire the object when is built.

## Tenant cluster creds

Without the CR we can't know which credentials to use (BYOC). So we can't just build these credentials and pass them around. They need to be built when reacting to a watched object. To build them, we need the CR to infer the credential secret.


So basically, if you don't care about which credentials are used for this particular call, just use the control plane ones. If you want to make the call with the tenant cluster credentials, your interface needs to either receive the credentials object, or the credential secret, or the CR.

Reply:
sounds good, can you link some place where we use this strategy?

Reply:
Not sure which one are you asking for. If you don't care about credentials and using the control plane is fine, maybe you can use this `AzureClientSet`? https://github.com/giantswarm/azure-operator/blob/5504970f61319f91a1e392295e85a53f79156331/service/service.go#L440

Pass it to your object in creation time and that's it.

Reply:
Yeah I was referring to the first strategy. Thanks

Reply:
should be done